src/boot/z_std_dma.c

```diff
@@ -351,19 +352,17 @@ s32 DmaMgr_SendRequestImpl(DmaRequest* req, u32 ram, u32 vrom, u32 size, u32 unk
     req->notifyQueue = queue;
     req->notifyMsg = msg;
 
-    if (1) {
-        if ((sDmaMgrQueueFullLogged == 0) && (sDmaMgrMsgQueue.validCount >= sDmaMgrMsgQueue.msgCount)) {
-            sDmaMgrQueueFullLogged++;
-            osSyncPrintf("%c", BEL);
-            osSyncPrintf(VT_FGCOL(RED));
-            osSyncPrintf("dmaEntryMsgQが一杯です。キューサイズの再検討をおすすめします。");
-            LOG_NUM("(sizeof(dmaEntryMsgBufs) / sizeof(dmaEntryMsgBufs[0]))", ARRAY_COUNT(sDmaMgrMsgs),
-                    "../z_std_dma.c", 952);
-            osSyncPrintf(VT_RST);
-        }
+    if (1 && (sDmaMgrQueueFullLogged == 0) && MQ_IS_FULL(&sDmaMgrMsgQueue)) {
+        sDmaMgrQueueFullLogged++;
+        osSyncPrintf("%c", BEL);
+        osSyncPrintf(VT_FGCOL(RED));
+        osSyncPrintf("dmaEntryMsgQが一杯です。キューサイズの再検討をおすすめします。");
+        LOG_NUM("(sizeof(dmaEntryMsgBufs) / sizeof(dmaEntryMsgBufs[0]))", ARRAY_COUNT(sDmaMgrMsgBuf), "../z_std_dma.c",
+                952);
+        osSyncPrintf(VT_RST);
     }
 
-    osSendMesg(&sDmaMgrMsgQueue, req, OS_MESG_BLOCK);
+    osSendMesg(&sDmaMgrMsgQueue, (OSMesg)req, OS_MESG_BLOCK);
```

Comment:
Casting a OSMesg to OSMesg seems unnecessary, but I guess this goes back to the question about always casting when calling the function or not, and I'm still not sure what makes the most sense

Reply:
`req` is a `DmaRequest*` here, which indeed is trivially castable to `OSMesg` aka `void*`, but it's not super obvious either especially with all the weird casts to `OSMesg` there are in various places imo

the cast happens regardless, it's just that making it explicit seems like a better idea? to me at least it does

Reply:
Oh I must have misread this and confused it with the msg variable, so yeah the cast makes a bit more sense.
I'm still on the fence about it because omitting it doesn't show up as a warning or anything and it's arguably useless, so it will likely be forgotten in the future (either in the repo itself, or in a modding scenario). But I guess it's fine either way

Reply:
I'm thinking explicit casts to `OSMesg`/`OSMesg*` when that's not already the type of the argument being passed, would really help confusion between

- `osSendMesg` taking in a message value `OSMesg` (`void*` but also used to pass integers for example https://github.com/zeldaret/oot/blob/a9084ad65e272441e838723a9ebade9f4b01d9eb/src/code/audio_load.c#L1711)
- `osRecvMesg` taking in a pointer to where to store a message, aka `OSMesg*` (`void**`)

The pointer of pointer usage especially on a `void` makes it imo confusing if you're passing a value (or "pointer value" I guess since the type for the message value `OSMesg` is `void*`) or a pointer (a pointer to the "pointer value")😅
but maybe there should be no unneeded casts at all and we should just be smarter